napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+However, it's the caller's responsibility to make sure that the 
+`napi_value` in question is of the JavaScript type expected by the API.
+
+
+## Functions
+
+### *napi_get_property_names*
+
+#### Signature
+```
+napi_status napi_get_property_names(napi_env env,
+                                    napi_value object,
+                                    napi_value* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to retrieve the properties
+- `[out] result`: A `napi_value` representing an array of JavaScript objects that represent the property names of the object. You can iterator over `result` using `napi_get_array_length` and `napi_get_element`.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_set_property*
+
+#### Signature
+```
+napi_status napi_set_property(napi_env env,
+                              napi_value object,
+                              napi_value key,
+                              napi_value value);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object on which to set the property
+- `[in]  key`: The name of the property to set
+- `[in]  value`: The property value
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_property*
+
+#### Signature
+```
+napi_status napi_get_property(napi_env env,
+                              napi_value object,
+                              napi_value key,
+                              napi_value* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to retrieve the property
+- `[in]  key`: The name of the property to retrieve
+- `[out] result`: The value of the property
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_has_property*
+
+#### Signature
+```
+napi_status napi_has_property(napi_env env,
+                              napi_value object,
+                              napi_value key,
+                              bool* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object to query
+- `[in]  key`: The name of the property whose existence to check
+- `[out] result`: Whether the property exists on the object or not
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_set_named_property*
+
+#### Signature
+```
+napi_status napi_set_named_property(napi_env env,
+                                    napi_value object,
+                                    const char* utf8Name,
+                                    napi_value value);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object on which to set the property
+- `[in]  utf8Name`: The name of the property to set
```

Comment:
Maybe mention in a Description section that the "named" property functions are equivalent to calling the non-"named" functions with a `napi_value` created from the name string.